src/renderer/services/MixpanelServices.ts

```diff
@@ -35,31 +38,36 @@ export const sendProgramToMix = (
       );
     }
     if (runProgram) {
-      sendEventToMix("Program Run", nodeList, "nodeList");
+      sendEventToMix(MixPanelEvents.programRun, nodeList, "nodeList");
     }
   }
 };
 
-export const sendEventToMix = (
-  event: string,
-  data: string,
-  dataType = "data",
+export const sendEventToMix = async (
+  event: MixPanelEvents | string,
+  data?: Record<string, unknown> | string,
+  dataType: string = "data",
 ) => {
+  if (await isCI()) return;
   if (enable) {
     try {
-      mixpanel.track(event, { [dataType]: data });
+      mixpanel.track(
+        event,
+        typeof data === "string" ? { [dataType]: data } : data,
+      );
```

Comment:
I feel like you could just only use `Record<string, unknown>` then make the caller specify the key instead of passing it as a separate parameter and converting it into a record in the function.

Reply:
I was initially considering backward compatibility to avoid altering existing implementations of this function in other places. However, I'll now proceed and implement this, as you suggested. Thanks!